darts/utils/likelihood_models/sklearn.py

```diff
@@ -412,59 +413,269 @@ def _get_median_prediction(self, model_output: np.ndarray) -> np.ndarray:
+                    "`component_names` requires the likelihood to be fitted "
+                    "but `ClassProbabilityLikelihood` is not fitted."
+                )
+            )
+        # format: <component_name>_p_<label>
+        return [
+            f"{component_name}_{parameter_name}"
+            for i, component_name in enumerate(input_series.components)
+            for parameter_name in self.parameter_names[
+                self._index_first_param_per_component[
+                    i
+                ] : self._index_first_param_per_component[i + 1]
+            ]
+        ]
+
+    def _estimator_predict(
+        self,
+        model,
+        x: np.ndarray,
+        **kwargs,
+    ) -> np.ndarray:
+        # list of length n_components * output_chunk_length of numpy arrays of shape (n_samples*n_series, n_classes)
+        model_output = model.model.predict_proba(x, **kwargs)
+        if not isinstance(model_output, list):
+            model_output = [model_output]
+
+        # shape: (output_chunk_length, n_series * n_samples, n_likelihood_parameters)
+        # n_likelihood_parameters is the sum of the number of classes for each component
+        class_proba = np.zeros((
+            self._n_outputs,
+            model_output[0].shape[0],
+            self.num_parameters,
+        ))
+
+        # ordered first by _n_outputs then by component: [comp0_chunk0, comp1_chunk0, ..]
+        n_component = len(model_output) // self._n_outputs
+        # filling the class_proba array with the predicted probabilities
+        for i, params_proba in enumerate(model_output):
+            component_index = i % n_component
+            output_index = i // n_component
+            class_proba[
+                output_index,
+                :,
+                self._index_first_param_per_component[
+                    component_index
+                ] : self._index_first_param_per_component[component_index + 1],
+            ] = params_proba
+
+        # shape (output_chunk_length, n_series * n_samples, n_likelihood_parameters)
+        # n_likelihood_parameters is the sum of the number of classes for each component
+        # n_likelihood_parameters are ordered as in component_names:
+        # [comp0_l1, comp0_l2, comp1_l1, comp1_l2, comp1_l3,...]
+        # should be accessed through _index_first_param_per_component (for comp1 => 2)
+        return class_proba
+
+    def sample(self, model_output: np.ndarray) -> np.ndarray:
+        """
+        Samples a prediction from the likelihood distribution and the predicted parameters.
+        """
+        # shape (output_chunk_length, n_series * n_samples, n_likelihood_parameters)
+        n_output, n_samples, _ = model_output.shape
+
+        n_component = len(self._classes) // self._n_outputs
+
+        # shape (output_chunk_length, n_series * n_samples, n_components)
+        preds = np.empty((n_output, n_samples, n_component), dtype=int)
+
+        for output_idx in range(n_output):
```

Comment:
as you mentioned this can be quite inefficient for large number of samples (it is expected for example that users predict with `num_samples=1000`).

Can we improve this? Maybe we could pass the `num_samples` to this method and then instead of picking one label per sample, we pick `num_samples` per series?